test/helpers/handshake.c

```diff
@@ -313,6 +313,15 @@ static int verify_reject_cb(X509_STORE_CTX *ctx, void *arg) {
     return 0;
 }
 
+static int verify_retry_cb(X509_STORE_CTX *ctx, void *arg) {
+    static int n_retries = 1;
```

Comment:
Hmm. Surely this means we can only do this test once? Any subsequent tests will have `n_retries` already decremented?

Reply:
Right. I though it would be sufficient to test `SSL_TEST_VERIFY_RETRY_ONCE` only once :wink:
Yet no problem, we can do the test more often.
I've also strengthened the test by checking that after each connection `verify_retry_cb` has been active as expected.